sp-sim/src/gimlet.rs

```diff
@@ -1107,12 +1115,18 @@ impl SpHandler for Handler {
         component: SpComponent,
     ) -> Result<u16, SpError> {
         warn!(
-            &self.log, "asked for component active slot (not supported for sim components)";
+            &self.log, "asked for component active slot";
             "sender" => %sender,
             "port" => ?port,
             "component" => ?component,
         );
-        Err(SpError::RequestUnsupportedForComponent)
+        if component == SpComponent::ROT {
+            Ok(rot_slot_id_to_u16(self.rot_active_slot))
+        } else if component == SpComponent::SP_ITSELF {
+            Ok(rot_slot_id_to_u16(self.sp_active_slot))
```

Comment:
The real SP returns `RequestUnsupportedForComponent` for anything other than the RoT for both this and `component_set_active_slot` - I think we could drop `self.sp_active_slot` and keep that same behavior here.